src/main/cpp/instruments/baseinstrument.cpp

```diff
@@ -66,6 +70,11 @@ std::vector<BaseAudioEvent*>* BaseInstrument::getEvents()
     return _audioEvents;
 }
 
+std::vector<BaseAudioEvent*>* BaseInstrument::getEventsForMeasure( int measureNum )
+{
+    return _audioEventsPerMeasure.size() <= measureNum ? nullptr : _audioEventsPerMeasure.at( measureNum );
+}
+
```

Comment:
Performance improvement: events are now indexed by the measures they occupy. This allows the sequencer to not loop over endless event lists (just to collect a handful of events that are eligible for playback at the current position) but instead retrieve them from the appropriate measure index.